### PR TITLE
Safe Compression Commit

### PR DESCRIPTION
**Pull request by ExeVirus**
Addresses #2803 concerns

Only a single program was used to safely compress all textures in MTG.

This was done using [optipng](https://sourceforge.net/projects/optipng/) with the following settings:

```
optipng -o7 -zm1-9 -nc -strip all -clobber %%f
```

The key command used is -nc, which stops color mode changes. I.e. any RGBA textures will remain RGBA textures, with none being changed to 8-bit indexed mode.

The resulting compression:

| type | size (Bytes) | % of original texture size |
| --- | --- | --- |
| Original | 247,571 | 100% |
| Compressed | 152,189 | 61.5% |
| Unsafe Compressed | 140,719 | 56.8% |

**Summary: Saves 93.14 KB**

### CODE REVIEW

**Comment by ExeVirus** [comment]
We may someday go back and attempt further compression by reducing bit depth, but since 5.4 release is Saturday, let's do the less controversial, safe commit.

**Comment by mlloewen** [comment]
You might want to check out oxipng found at [https://github.com/shssoichiro/oxipng](https://github.com/shssoichiro/oxipng).  In short it is a rewrite of optipng in rustlang with some enhancements(multithread etc).

**Comment by rubenwardy** [comment]
Hi,

please may you add a file `optimize_textures.sh` with the following content:

```bash
#!/bin/bash

# Colors with 0 alpha need to be preserved,
# because opaque leaves ignore alpha.

find -name '*.png' -print0 | xargs -0  optipng -o7 -zm1-9 -nc -strip all -clobber
```

**Comment by sfan5** [comment]
> Why are index color modes a problem?

See https://github.com/minetest/minetest/blob/master/doc/texture_packs.txt#L226-L232 or https://github.com/minetest/minetest_game/pull/2803#issuecomment-756321696

**Comment by ExeVirus** [comment]
Ruben, what do you mean by "add a File" I can do so easily, but I don't know where to put it.

**Comment by rubenwardy** [comment]
> Ruben, what do you mean by "add a File" I can do so easily, but I don't know where to put it.

I was referring to the root of this repository. A `utils` folder would make sense if there were more than one utility script

**Comment by ExeVirus** [comment]
Okay, rebased and added _optimize_textures.sh_ as written by Rubenwardy.
Tested the script on this PR on Lubuntu 20.04 LTS, same result as PR.

**Review by rubenwardy — APPROVED** [review]

File: optimize_textures.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+# Colors with 0 alpha need to be preserved, because opaque leaves ignore alpha.
+# For that purpose, the use of indexed colors is disabled (-nc).
```
Comment (sfan5) [inline optimize_textures.sh:4]: Please put this in an "utils" folder in case an .obj compression script is added in the future.

Comment (rubenwardy) [reply optimize_textures.sh:4]: This is fine by me

Comment (ExeVirus) [reply optimize_textures.sh:4]: Okay, will do. This brings up the question: should I go ahead and add the compress.obj script to the same utils folder in #2807? Then, should I add a readme.md to that folder saying how to use both scripts in a new PR?

**Comment by ExeVirus** [comment]
Okay, conflicts dealt with, changed the script directory to "utils"

**Review by SmallJoker — APPROVED** [review]
Alpha contents are preserved. No texture format warnings. Looks good.

**Pull request merged**